### PR TITLE
Ads cleanup

### PR DESCRIPTION
**Pull request by infinite-persistence**
#1018

- [x] see why it doesn't show on mobile + homepage.
- [x] show on category pages (make sure it works on mobile)
- [x] show on channel pages (we have it in the tile view, not list view)
    - [x] It's now shown on both layouts, but the list version still looks like tile in mobile.  Needs some more CSS help from Rave.
- [x] "total blocking time" improvements in homepage (from 2s down to ~500ms)
    - image
- [x] fixes double ads when language is changed via the first-time-use nag.
- [x] fixes ads not resized correctly when moving from small to medium/large, and vice-versa.

I kinda like the original's (Recommended's) background color, to differentiate it against the rest.  Can remove if dislike.

### CODE REVIEW

**Comment by mayeaux** [comment]
Screen Shot 2022-03-08 at 3 56 48 PM
Screen Shot 2022-03-08 at 3 56 33 PM

Matter of personal preference which is better, personally I say the one without the shaded background
Screen Shot 2022-03-08 at 4 01 30 PM

With the shaded background it's a bit too jarring since it pulls the attention that way and also the video sizes don't line up, I think it being the only moving thing on the homepage is enough in terms of attention grabbing, otherwise it makes the user start at the bottom right of the page to look at an ad rather than to start at the top left and the notice the ad later at some point. Not a blocker though I don't think (but I am just worried about impact on the homepage esp since most people are just getting the meteor 'ad' which we don't get compensated for, and I don't think we are compensated for clickthroughs on that ad either). If it was a case where the ad was being served for most people, or we could otherwise hide it if it it wasn't being filled, there would be a stronger argument to have it displayed this way, that's my take.

**Review by mayeaux — APPROVED** [review]

**Comment by tzarebczan** [comment]
Only issue I found was uneven tile loading - we used to have this before the redesign too, and it can be fixed later. The last row filled in right after I took the screenshot. 
image

File: web/component/ads/view.jsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ function Ads(props: Props) {
     return (
       <div
         className={classnames('ads ads__claim-item', className, {
-          'ads__claim-item--tile': tileLayout,
+          'ads__claim-item--tile': tileLayout, // with no tileLayout it indicates sidebar ad
```
Comment (infinite-persistence) [inline web/component/ads/view.jsx:124]: This comment is wrong.  Ads in Category Page and Channel Page can be either Tile or List layout.
This ad component is general purpose and not tied to sidebar.

File: ui/effects/use-last-visible-item.js [hunk 2]
```diff
@@ -18,8 +24,11 @@ export default function useLastVisibleItem(injectedItem: ?InjectedItem, listRef:
       const timer = setTimeout(() => {
         if (listRef.current) {
           const screenBottom = window.innerHeight;
+
+          // claim preview tiles
```
Comment (infinite-persistence) [inline ui/effects/use-last-visible-item.js:28]: Can be list, tiles, or even non ClaimPreview entirely. It's self-documenting:  items = all items in the listRef

File: ui/effects/use-last-visible-item.js [hunk 3]
```diff
@@ -4,6 +4,12 @@ import type { Node } from 'react';
 
 type InjectedItem = { node: Node, index?: number, replace?: boolean };
 
+/**
+ * Returns the index indicating where in the claim-grid to inject the ad element
+ * @param injectedItem
+ * @param listRef - A reference to the claim-grid
```
Comment (infinite-persistence) [inline ui/effects/use-last-visible-item.js:10]: Misleading.  In List mode, it's not even `claim-grid` ... it's a section.  
Anything that houses a bunch of `<li>` can be listRef.

**Review by infinite-persistence — COMMENTED** [review]
Comments are incorrect/misleading/redundant.

**Pull request merged**